blocks/key-facts/key-facts.js

```diff
@@ -6,40 +9,95 @@ function stripEmptyTags(main, child) {
   }
 }
 
-export default function decorate(block) {
+const blockName = 'key-facts';
+const variantClasses = ['trailing-line'];
+
+const keyFactsColumns = (el, v2) => {
+  el.classList.add(`${blockName}__key-item`);
+
+  // adding icon
+  const icon = el.querySelector('i, .icon');
+
+  if (icon) {
+    el.prepend(icon);
+    icon.classList.add(`${blockName}__icon`);
+  }
+
+  // clearing empty paragraphs
+  const paragraphs = el.querySelectorAll('p');
+  paragraphs.forEach((paragraph) => {
+    stripEmptyTags(el, paragraph);
+  });
+
+  const preSubheading = el.querySelector(':scope > :not(strong, i):first-child');
+  preSubheading?.classList.add('subtitle-2');
+
+  // find and split number/unit
+  const value = el.querySelector('strong:only-child');
+  // separate number and unit. e.g. up | +31% | Freight efficiency, or 120,000 | psi,
+  const [, prenumber, number, postnumber, text] = value.innerHTML.match('([\\+,\\-]*)([0-9,\\.,]+)([\\%]*) *(.*)');
```

Comment:
Nice to have: testing with test cases :)

Reply:
Well in my opinion it will be nice to get rid of the regexp :)
But we don't know on which pages it is used.